### ISSUE TITLE
Router.navigate executes although trigger:false with encoded hash fragment in Firefox

### ISSUE THREAD

**Issue by bobmop** [jashkenas/backbone#3941]
`Router.navigate()` will execute with `trigger:false` if the has fragment will contain url encoded characters in FireFox (tested with V39, V43).

correct behaviour in Backbone 1.1.2: https://jsfiddle.net/g9zmwmdk/10/
wrong behaviour in Backbone 1.2.3: https://jsfiddle.net/g9zmwmdk/9/

I thinkl the new behaviour was introduced in https://github.com/jashkenas/backbone/compare/1.1.2...1.2.0#diff-0d56d0d310de7ff18b3cef9c2f8f75dcR1764

**Comment by akre54** [comment]
Yup, specifically in #3434 as a fix for #3426. I'm not super familiar with this code but I think the solution is to not to pre-encode your string. See if that works for you?

**Comment by bobmop** [comment]
Thank you for the feedback.

I get a path from an api which is already encoded and navigate() using this path without extra encoding.

I think i found the "broken" code:

After `navigate()` is called correctly, `Backbone.History.checkUrl()` is called by an event listener and will compare the cached (decoded) fragment with `getFragment()`, which is not decoded by Backbone itself. The result is another fragment than the cached one and `loadUrl()` including the configured handler is executed.

I think Backbone has to be fixed, e.g. calling this.decodeFragment() in this.getFragment().

**Comment by akre54** [comment]
If you want to open a pull with your proposed solution and some tests that fail on master we'd be happy to take a look.

**Comment by bobmop** [comment]
here is a proposal including a test: #3955
